### ISSUE TITLE
Coder ignoring default variable defined in the variables.tf for the template

### ISSUE THREAD

**Issue by dmildh-absci** [coder/coder#12133]
Hello we are using coder installed via helm 2.8.2 am seeing some strangeness with the template updates in relation to default vars.  For example within a Kubernetes template for workspace we have a varibles.tf and have the image tag for the workspaces default like so:

```
variable "coder_image" {
  type        = string
  default     = "someimage:123456"
  description = "Coder image to use for workspace"
}
```

Inside the main.tf we are using the var.coder_image.

The first push of the template is fine and the correct image is used, however if we update the default value the changes do not seem to propagate unless we pass the -var coder_image="newimage:tag" on the cli 

```
coder templates push --directory ./kubernetes --var coder_image=newimage:tag
```

**Comment by mtojek** [comment]
TL;DR I verified the logic, it works as intended.

Template variable requires a value to be provided on the first push when the variable is added. Template admin has two options:
1. Do not define `default` property, and use `--var  name=value` command-line argument.
2. Define `default` property for the template variable, and let Coder use it if admin doesn't input it via CLI.

Once the template is pushed, variables are stored in the database table. They are linked to specific template version, and you can edit them via _Template Settings_ without an extra push. Coder templates do not operate on `default_value`s, but on resolved `value`s on push.

Q: Why does it work this way?
A: Some users prefer `default_value`s not to override their vars settings accidentally on push. That's it.

Q: How can we solve my case?
A: You can create a file called `terraform.tfvars` and drop it in the template directory:

```
coder_image = newimage:tag
``` 

Coder will use it to override the Template setting for variables. Of course, you can keep this file in the repository.

Anyway, I'm going to improve documentation around this issue as it looks that we haven't described the concept of `.tfvars`.

**Comment by dmildh-absci** [comment]
Thank you @mtojek It would also be nice also if the terraform.tfvars values showed up in the source code in the console somewhere.

**Comment by mtojek** [comment]
`terraform.tfvars` are not uploaded to Coder. They are just interpreted on CLI level.
